src/application/trading-copilot/subscriptions-management/components/subscriptions-list/subscriptions-list.tsx

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from 'react';
+
+import { Empty, Spinner } from 'shared/ui';
+
+import { Subscription } from '../../../types';
+
+import { SubscriptionItem } from './subscription-item';
+
+type Properties = {
+  subscriptions: Subscription[];
+  subscriptionsLoading: boolean;
+  subscriptionsUpdatePending: boolean;
+  className?: string;
+  onRemove: (ensName: Subscription['ensName']) => void;
+};
+
+export const SubscriptionsList = ({
+  subscriptions,
+  subscriptionsLoading,
+  subscriptionsUpdatePending,
+  onRemove,
+  className,
+}: Properties) => {
+  const subscriptionsListBody = useMemo(() => {
+    if (subscriptionsLoading) {
+      return (
+        <Spinner className="mt-14 flex w-full items-center justify-center" />
+      );
+    }
+
+    if (subscriptions.length === 0) {
+      return (
+        <Empty text="Your subscriptions list is empty" className="mt-14" />
+      );
+    }
+
+    return (
+      <div className="relative">
+        <ul className="space-y-2">
+          {subscriptions.map((subscription) => {
+            return (
+              <SubscriptionItem
+                subscription={subscription}
+                key={subscription.ensName}
+                onRemove={onRemove}
+              />
+            );
+          })}
+        </ul>
+        {subscriptionsUpdatePending && (
+          <Spinner className="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2" />
+        )}
+      </div>
+    );
+  }, [
+    onRemove,
+    subscriptions,
+    subscriptionsLoading,
+    subscriptionsUpdatePending,
+  ]);
```

Comment:
I think it should be composed inside `subscriptions-management.tsx` so you don't need to pass query details and handle things like loading or empty list, you can use the advantage of compound components pattern and it would look like so:
```ts
const subscriptions = subscriptionsQuery.data ?? [];
const hasSubscriptions = subscriptions.length > 0;

<SubscriptionForm onSubmit={handleAddNewSubscription} />
<div className="mt-4">
  <SubscriptionsList.Title>
    Your Subscriptions
  </SubscriptionsLsit.Title>

  {subscriptionsQuery.isLoading && <SubscriptionsList.Loading />
  {!hasSubscriptions && <SubscriptionsList.Empty />}
  {hasSubscriptions && <SubscriptionsList subscriptions={subscriptions} />}
</div>
```